font-types/src/tag.rs

```diff
@@ -233,6 +232,42 @@ impl Default for Tag {
     }
 }
 
+// fancy impls: these will serialize to a string if the target format is
+// human-readable, but to bytes otherwise.
+//
+// NOTE: this means that tags which are not utf-8 will fail to serialize to
+// json/yaml.
+#[cfg(feature = "serde")]
+impl serde::Serialize for Tag {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            std::str::from_utf8(&self.0)
+                .map_err(serde::ser::Error::custom)?
+                .serialize(serializer)
```

Comment:
Optional and IMHO can wait until we have a real example: you could fallback to the non-string path if string fails. Deserialize would similarly attempt string, presumably failing if the initial serialize failed, and fallback to parsing `[u8; 4]`.

I suppose you could stuff it into an enum to serialize/deserialize to avoid deserialize having to attempt multiple formats.

Reply:
I looked into that and while possible it's kind of annoying, and so I ended up reaching the same conclusion you have here, e.g. let's wait and see if this is ever actually an issue.